cf-monitord/env_monitor.c

```diff
@@ -556,7 +557,10 @@ static void PublishEnvironment(Item *classes)
 
     unlink(ENVFILE_NEW);
 
-    if ((fp = fopen(ENVFILE_NEW, "a")) == NULL)
+    const mode_t old_umask = SetUmask(0077);
+    FILE *fp = safe_fopen(ENVFILE_NEW, "a");
```

Comment:
`FILE *fp;` five lines above has to be dropped.

Reply:
ACK